### PR TITLE
HPCC-32890 Add cleanjobqueues to daliadmin

### PR DESCRIPTION
**Pull request by jakesmith**
cleanjobqueues scans all /JobQueues and the Clients within them, and deletes any which are not in use (connected==0).

NB: this should only be necessary on old (pre 9.4 systems). (see HPCC-31807 in 9.4+)

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/daliadmin/daadmin.cpp [hunk 1]
```diff
@@ -3385,4 +3385,43 @@ void removeOrphanedGlobalVariables(bool dryrun, bool reconstruct)
...
+    {
+        WARNLOG("Failed to connect to /JobQueues");
+        return;
+    }
+    Owned<IPropertyTreeIterator> queueIter = conn->queryRoot()->getElements("Queue");
+    ForEach(*queueIter)
+    {
+        IPropertyTree &queue = queueIter->query();
+        const char *name = queue.queryProp("@name");
+        if (isEmptyString(name)) // should be blank, but guard
```
Comment (shamser) [inline dali/daliadmin/daadmin.cpp:3401]: comment: should **not** be blank ?

File: dali/daliadmin/daadmin.cpp [hunk 2]
```diff
@@ -3385,4 +3385,43 @@ void removeOrphanedGlobalVariables(bool dryrun, bool reconstruct)
...
+            IPropertyTree &client = clientIter->query();
+            if (client.getPropInt("@connected") == 0)
+                toRemove.push_back(&client);
+        }
+        if (!dryRun)
+        {
+            for (auto &client: toRemove)
+                queue.removeTree(client);
+        }
+        PROGLOG("%s %u stale client entries", dryRun ? "Dryrun, there are" : "Removed", (unsigned)toRemove.size());
```
Comment (shamser) [inline dali/daliadmin/daadmin.cpp:3421]: minor: would it be useful to show the queue name too?

**Review by shamser — COMMENTED** [review]
Minor: a code comment needs to be changed.

**Review by shamser — CHANGES_REQUESTED** [review]
Minor: a code comment needs to fixed.

File: dali/daliadmin/daadmin.cpp (hunk 2 above)
Comment (jakesmith) [reply dali/daliadmin/daadmin.cpp]: yeah, changed.

**Comment by jakesmith** [comment]
@shamser - please see 2nd commit.

**Review by shamser — APPROVED** [review]

**Review by ghalliday — APPROVED** [review]

**Pull request merged**